src/http/proxy-aware-client.ts

```
@@ -0,0 +1,422 @@
+			const requestModule = proxy.protocol === 'https' ? httpsRequest : httpRequest;
+
+			const baseHeaders: OutgoingHttpHeaders = {
+				Host: targetUrl.host,
+			};
+
+			if (proxy.username !== undefined && proxy.password !== undefined) {
+				baseHeaders['Proxy-Authorization'] = `Basic ${Buffer.from(`${proxy.username}:${proxy.password}`).toString('base64')}`;
+			}
+
+			// Convert RequestInit headers to OutgoingHttpHeaders
+			const requestHeaders: OutgoingHttpHeaders = { ...baseHeaders };
```

ESLint

[error] Unsafe assignment of an `any` value.

(ts/no-unsafe-assignment)